app/main.py

```diff
@@ -1,34 +1,160 @@
+
+            self.field[tuple(cells)] = Ship(start, end)
+
+        self._validate_field()
+
+    def fire(self, location: tuple) -> str:
+        for cells, ship in self.field.items():
+            if location in cells:
+                row, column = location
+                ship.fire(row, column)
+                if ship.is_drowned:
+                    return "Sunk!"
+                return "Hit!"
+        return "Miss!"
+
+    def print_field(self) -> None:
+        print()
+        for row in range(10):
+            line = ""
+            for column in range(10):
+                symbol = self.get_symbol(row, column)
+                line += symbol
+            print(f"{line}")
+
+    def get_symbol(self, row: int, column: int) -> str:
+        for cells, ship in self.field.items():
+            if (row, column) in cells:
+                if ship.is_drowned:
+                    return "X"
+                deck = ship.get_deck(row, column)
+                if deck.is_alive:
+                    return u"\u25A1"
+                return "*"
+        return "~"
+
+    def _validate_field(self) -> None:
+
+        if len(self.field) != 10:
+            raise _FieldError("The total number of the ships should be 10")
+
+        ships = {
+            1: 0,
+            2: 0,
+            3: 0,
+            4: 0,
+        }
+        for cells in self.field:
+            ships[len(cells)] += 1
+        if ships[4] != 1:
+            raise _FieldError("There should be 4 single-deck ships")
+        if ships[3] != 2:
+            raise _FieldError("There should be 3 double-deck ships")
+        if ships[2] != 3:
+            raise _FieldError("There should be 2 three-deck ships")
+        if ships[1] != 4:
+            raise _FieldError("There should be 1 four-deck ship")
+
+        empty_cells = []
+        for cells in self.field:
+            for cell in cells:
+                row, column = cell
+                self._add_new_cell(cells, empty_cells, row + 1, column)
+                self._add_new_cell(cells, empty_cells, row - 1, column)
+                self._add_new_cell(cells, empty_cells, row - 1, column + 1)
+                self._add_new_cell(cells, empty_cells, row, column + 1)
+                self._add_new_cell(cells, empty_cells, row + 1, column + 1)
+                self._add_new_cell(cells, empty_cells, row + 1, column - 1)
+                self._add_new_cell(cells, empty_cells, row, column - 1)
+                self._add_new_cell(cells, empty_cells, row - 1, column - 1)
+        empty_cells_set = set(empty_cells)
+        for cells in self.field:
+            for cell in cells:
+                row, column = cell
+                if (row, column) in empty_cells_set:
+                    raise _FieldError("Ships shouldn't "
+                                      "be located in the neighboring cells")
+
+    @staticmethod
+    def _add_new_cell(
+            cells: tuple,
+            checked_cells: list,
```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]